kaggle_environments/envs/llm_20_questions/llm_20_questions.py

```diff
@@ -75,6 +75,66 @@ def answerer_agent(obs):
 
 agents = {"guesser": guesser_agent, "answerer": answerer_agent}
 
+def guesser_action(active, inactive, step):
+    guessed = False
+    if not active.action:
+        active.status = "ERROR"
+    elif active.observation.turnType == "ask":
+        active.observation.questions.append(active.action)
+        inactive.observation.questions.append(active.action)
+    elif active.observation.turnType == "guess":
+        active.observation.guesses.append(active.action)
+        inactive.observation.guesses.append(active.action)
+    if active.action and keyword_guessed(active.action):
+        guessed = True
+        score = 20 - int(step / 3)
+        active.reward = score
+        inactive.reward = score
+        active.status = "DONE"
+        inactive.status = "DONE"
+        active.observation.keyword = keyword
+        active.observation.category = category
+    inactive.observation.keyword = keyword
+    inactive.observation.category = category
+    return guessed
+
+def answerer_action(active, inactive):
+    active.observation.keyword = keyword
+    active.observation.category = category
+    response = active.action
+    if not response:
+        response = "none"
+        active.status = "ERROR"
+    elif response.lower().__contains__("yes"):
+        response = "yes"
+    elif response.lower().__contains__("no"):
+        response = "no"
+    else:
+        response = "maybe"
+        active.status = "ERROR"
+    active.observation.answers.append(response)
+    inactive.observation.answers.append(response)
+
+def increment_turn(active, inactive, step, guessed):
+    if step == 59 and not guessed:
+        active.observation.keyword = keyword
```

Comment:
Since you're doing all the same things here for both `active` and `inactive`, could consider pulling out a helper function